procrastinate/job_context.py

```diff
@@ -80,11 +82,4 @@ def job_description(self, current_timestamp: float) -> str:
         return message
 
     def should_abort(self) -> bool:
-        assert self.job.id
-        job_id = self.job.id
-        return self.app.job_manager.get_job_abort_requested(job_id)
-
-    async def should_abort_async(self) -> bool:
-        assert self.job.id
-        job_id = self.job.id
-        return await self.app.job_manager.get_job_abort_requested_async(job_id)
```

Comment:
What about extending the `Job` class instead to have `abort_requested`. This way, the existing `list_job` operation covers it and there is no need to have distinct operations for every piece of state?

Reply:
Seems to make sense. I agree!

Reply:
Ah, I might not have understood your proposal.

I think it's worth keeping a `should_abort_async` on the context just because we advertised it before because it's really not going to get in the way of maintenance, independently from how we can make the info available on the job. It's literally:

```python
async def should_abort_async(self):
    # As it stands, `should_abort` doesn't do I/O as of now.
    return self.should_abort()
```

I don't see a case where this get in our way.

Reply:
Given how short lived that function is, how simple it is to migrate to the non async version and the opportunity to have breaking changes as part of v3, it seems reasonable to remove it.

Reply:
Ok, won't fight you on this.